File: backend/entity/Post.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type Post struct {
	gorm.Model
	Descript           string          `json:"Descript"`
	Start              time.Time       `json:"Start"`
	End                time.Time       `json:"End"`
	Price              int             `json:"Price"`
	TypeID             uint            `json:"TypeID"`
	Type               Type            `gorm:"references:ID" json:"Type"`
	Service_UserID     uint            `json:"Service_UserID"`
	ServiceUser        ServiceUser     `gorm:"references:ID" json:"ServiceUser"`
	Service_ProviderID uint            `json:"Service_ProviderID"`
	ServiceProvider    ServiceProvider `gorm:"references:ID" json:"ServiceProvider"`
	StatusID           uint            `json:"StatusID"`
	Status             Status          `gorm:"references:ID" json:"Status"`
	Note               string          `json:"Note"`
}

File: frontend/src/models/Post.ts
import { StatusInterface } from "./Status";
import { TypeInterface } from "./Type";
import { ServiceProviderInterface, ServiceUserInterface } from "./User";

// Show Post
export interface PostSInterface {
    ID?: number;
    Descript?: string;
    Start: Date;
    End: Date;
    Price?: number;
    TypeID?: number;
    Type?: TypeInterface;
    Service_UserID?: number
    ServiceUser: ServiceUserInterface;
    Service_ProviderID?: number;
    ServiceProvider?: ServiceProviderInterface;
    StatusID?: number;
    Status?: StatusInterface;
    Note?: string;
}

// Create Post
export interface PostCInterface {
    ID?: number;
    Descript?: string;
    Start?: Date;
    End?: Date;
    Price?: number;
    TypeID?: number;
    Service_ProviderID?: number;
    StatusID?: number;
}

// Edit Post
export interface PostEInterface {
    ID?: number;
    Descript?: string;
    Start?: Date;
    End?: Date;
    Price?: number;
    TypeID?: number;
    Note?: string;
    Service_UserID?: number;
}